### PR TITLE
Expose new tile service

### PR DESCRIPTION
**Pull request by danpat**
This exposes the new `tile` plugin in `osrm-backend`.  See https://github.com/Project-OSRM/osrm-backend/pull/2000 for more details.

### CODE REVIEW

**Comment by willwhite** [comment]
Is this ready to merge or waiting for https://github.com/Project-OSRM/osrm-backend/pull/2000 to merge?

**Comment by TheMarex** [comment]
This should also unwrap the `tile()` response from the enclosing json and only return the pbf. Needs the corresponding PR in osrm-backend.

File: src/node_osrm.cpp [hunk 1]
```diff
@@ -573,6 +575,48 @@ void Engine::nearest(const Nan::FunctionCallbackInfo<v8::Value> &args)
...
+    if (args.Length() < 2)
+    {
+        Nan::ThrowTypeError("two arguments required");
+        return;
+    }
+
+    v8::Local<v8::Object> obj = Nan::To<v8::Object>(args[0]).ToLocalChecked();
+
+    v8::Local<v8::Value> x = obj->Get(Nan::New("x").ToLocalChecked());
+    if (!x->IsNumber() && !x->IsUndefined())
```
Comment (daniel-j-h) [inline src/node_osrm.cpp]: Related to https://github.com/Project-OSRM/osrm-backend/issues/2009: this does not catch nans and infs, does it?

Should we handle them on the API level?

**Comment by willwhite** [comment]
@springmeyer any idea why the tests in the pr travis test are failing? I looks like it might be trying to publish for both the push and the pr.

**Comment by TheMarex** [comment]
@willwhite yes that was one of the reason we disabled builds for both branches and PRs.

**Pull request merged**